JOSESwift/Sources/JWK.swift

```diff
@@ -83,4 +85,25 @@ public protocol JWK: Codable {
     /// - Returns: The JSON representation of the JWK as `Data` or
     ///            `nil` if the encoding failed.
     func jsonData() -> Data?
+
+    /// Generate the thumbprint based on hash function SHA256
+    ///  See [RFC-7638, Section 3.2](https://tools.ietf.org/html/rfc7638#section-3.2)
+    ///
+    /// - Returns: The thumbprint from the required members of the JWK key.
+    /// - Throws: A `JOSESwiftError` indicating any errors.
+    @available(iOS 11.0, *)
+    func thumbprint(algorithm: ThumbprintAlgorithm) throws -> String
+
+    @available(iOS 11.0, *)
+    func withKeyIdFromThumbprint(algorithm: ThumbprintAlgorithm) throws -> Self
```

Comment:
Maybe something like this would be clearer: 

```suggestion
    func withThumbprintAsKeyId(algorithm: ThumbprintAlgorithm) throws -> Self
```

What do you think?

Reply:
Yes, totally agree :)